### PR TITLE
fix: backport e2e tests

### PR DESCRIPTION
**Pull request by shotonoff**
## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: light/client.go [hunk 1]
```diff
@@ -659,12 +662,15 @@ func (c *Client) lightBlockFromPrimary(ctx context.Context) (*types.LightBlock,
...
-	l, err := c.primary.LightBlock(ctx, height)
+	l, err := c.getLightBlock(ctx, c.primary, height)
 	c.providerMutex.Unlock()
 
 	switch err {
 	case nil:
-		// Everything went smoothly. We reset the lightBlockRequests and return the light block
+
+	// catch canceled contexts or deadlines
+	case context.Canceled, context.DeadlineExceeded:
```
Comment (lklimek) [inline light/client.go:672]: Does this reordering  change anything?

File: light/client.go [hunk 2]
```diff
@@ -873,11 +888,3 @@ func (c *Client) compareFirstHeaderWithWitnesses(ctx context.Context, h *types.S
 	// remove all witnesses that misbehaved
 	return c.removeWitnesses(witnessesToRemove)
 }
-
-// providerShouldBeRemoved analyzes the nature of the error and whether the provider
-// should be removed from the light clients set
-func (c *Client) providerShouldBeRemoved(err error) bool {
```
Comment (lklimek) [inline light/client.go:879]: This function is used in light/detector.go

File: test/e2e/Makefile [hunk 3]
```diff
@@ -7,21 +6,21 @@ docker:
...
 
 node:
-	go build -o build/node -tags badgerdb,boltdb,cleveldb,rocksdb ./node
+	go build -gcflags="all=-N -l" -o build/node -tags badgerdb,boltdb,cleveldb,rocksdb ./node
 
 e2e/app/compile:
 	docker run --rm -it --entrypoint "/src/tenderdash/test/e2e/entrypoint.sh" -w "/src/tenderdash/test/e2e" -v $(ROOT_PATH):/src/tenderdash tenderdash/e2e-node
 
-runner/simple: runner e2e/app/compile
+runner/simple: runner tests e2e/app/compile
```
Comment (lklimek) [inline test/e2e/Makefile:14]: why are you running tests here?

File: test/e2e/networks/dashcore.toml [hunk 4]
```diff
@@ -24,17 +25,20 @@ validator04 = 100
 [validator_update.1010]
 validator01 = 100
 validator02 = 100
+validator03 = 0
```
Comment (lklimek) [inline test/e2e/networks/dashcore.toml:28]: Please double-check that all changes here are correct. Validator updates are tested in rotate.toml.

File: test/e2e/networks/dashcore.toml [hunk 5]
```diff
@@ -69,42 +73,25 @@ perturb = ["pause"]
...
 perturb = ["restart"]
 
 [node.light01]
 mode= "light"
-start_at= 1005
+start_at= 1010
 privval_protocol = "dashcore"
 persistent_peers = ["validator01", "validator02", "validator03"]
-
-[node.light02]
```
Comment (lklimek) [inline test/e2e/networks/dashcore.toml:106]: should we restore light nodes?

File: test/e2e/runner/rpc.go [hunk 6]
```diff
@@ -13,7 +13,7 @@ import (
 )
 
 const (
-	WaitForHeightTimeout = 20 * time.Second
+	WaitForHeightTimeout = 30 * time.Second
```
Comment (lklimek) [inline test/e2e/runner/rpc.go:16]: is this still needed or added for debugging?

**Review by lklimek — CHANGES_REQUESTED** [review]

File: crypto/xchacha20poly1305/xchachapoly_test.go [hunk 7]
```diff
@@ -25,19 +25,19 @@ func TestRandom(t *testing.T) {
 		plaintext := make([]byte, pl)
 		_, err := crand.Read(key[:])
 		if err != nil {
-			t.Errorf("error on read: %w", err)
+			t.Errorf("error on read: %v", err)
```
[the remark this answers is not in the document]
Comment (shotonoff) [reply crypto/xchacha20poly1305/xchachapoly_test.go]: for the errors %s and %v are identical

File: internal/statesync/reactor.go [hunk 8]
```diff
@@ -1059,12 +1059,12 @@ func (r *Reactor) initStateProvider(ctx context.Context, chainID string, initial
 			providers[idx] = NewBlockProvider(p, chainID, r.dispatcher)
 		}
 
-		r.stateProvider, err = NewP2PStateProvider(ctx, chainID, initialHeight, providers, r.paramsCh.Out, spLogger, r.dashCoreClient)
+		r.stateProvider, err = NewP2PStateProvider(ctx, chainID, initialHeight, r.cfg.TrustHeight, providers, r.paramsCh.Out, spLogger, r.dashCoreClient)
```
[the remark this answers is not in the document]
Comment (shotonoff) [reply internal/statesync/reactor.go:1062]: @QuantumExplorer do you have any concerns? 
remind the context:
* initial-height is the height when the network starts, like `1000`
* trust-height is the height when the node will start, like `1015`

File: light/client.go (hunk 1 above)
Comment (shotonoff) [reply light/client.go:672]: I did it 'cause upstream has the same order (just sync)

File: test/e2e/Makefile (hunk 3 above)
Comment (shotonoff) [reply test/e2e/Makefile:14]: this network presents in upstream, we keep it here for consistency

File: test/e2e/networks/dashcore.toml (hunk 5 above)
Comment (shotonoff) [reply test/e2e/networks/dashcore.toml:106]: the idea is to have the same configuration with upstream

File: test/e2e/runner/rpc.go (hunk 6 above)
Comment (shotonoff) [reply test/e2e/runner/rpc.go]: 20 sec was not enough for passing tests

**Review by shotonoff — COMMENTED** [review]
done, answered on the questions

File: test/e2e/log.sh [hunk 9]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env sh
```
Comment (lklimek) [inline test/e2e/log.sh:1]: do we want to store this file in git? or it's just a temporary testing script?

File: test/e2e/run.sh [hunk 10]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env sh
```
Comment (lklimek) [inline test/e2e/run.sh:1]: do we want to store this file in git? or it's just a temporary testing script?

File: test/e2e/stat.sh [hunk 11]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
```
Comment (lklimek) [inline test/e2e/stat.sh:1]: Do we want to store this file in git? or it's just a temporary testing script?

File: types/validator_address.go [hunk 12]
```diff
@@ -28,7 +28,7 @@ type ValidatorAddress struct {
 
 var (
 	// ErrNoHostname is returned when no hostname is set for the validator address
-	ErrNoHostname = errors.New("no hostname")
+	ErrNoHostname error
```
Comment (lklimek) [inline types/validator_address.go:31]: Why this change? It means `ErrNoHostname` is nil

**Review by lklimek — APPROVED** [review]

File: test/e2e/log.sh (hunk 9 above)
Comment (shotonoff) [reply test/e2e/log.sh]: this file and others was committed by accident

File: test/e2e/run.sh (hunk 10 above)
Comment (shotonoff) [reply test/e2e/run.sh]: it will be removed

File: test/e2e/stat.sh (hunk 11 above)
Comment (shotonoff) [reply test/e2e/stat.sh]: it will be removed

File: types/validator_address.go (hunk 12 above)
Comment (shotonoff) [reply types/validator_address.go]: revert

**Review by shotonoff — COMMENTED** [review]
clean up the files, revert a few changes
